### PR TITLE
DAOS-12256 common: fix ifv_rec_string()

### PR DESCRIPTION
**Pull request by NiuYawei**
Fix ifv_rec_string() & iv_rec_string() to get btr_record correctly
when the node is an interior node for direct key tree.

Required-githooks: true

Signed-off-by: Niu Yawei <yawei.niu@intel.com>

### Before requesting gatekeeper:

* [ ] Two review approvals and any prior change requests have been resolved.
* [ ] Testing is complete and all tests passed or there is a reason documented in the PR why it should be force landed and forced-landing tag is set.
* [ ] `Features:` (or `Test-tag*`) commit pragma was used or there is a reason documented that there are no appropriate tags for this PR.
* [ ] Commit messages follows the guidelines outlined [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments).
* [ ] Any tests skipped by the ticket being addressed have been run and passed in the PR.

### Gatekeeper:

* [ ] You are the appropriate gatekeeper to be landing the patch.
* [ ] The PR has 2 reviews by people familiar with the code, including appropriate watchers.
* [ ] Githooks were used. If not, request that user install them and check copyright dates.
* [ ] Checkpatch issues are resolved.  Pay particular attention to ones that will show up on future PRs.
* [ ] All builds have passed.  Check non-required builds for any new compiler warnings.
* [ ] Sufficient testing is done. Check feature pragmas and test tags and that tests skipped for the ticket are run and now pass with the changes.
* [ ] If applicable, the PR has addressed any potential version compatibility issues.
* [ ] Check the target branch.   If it is master branch, should the PR go to a feature branch?  If it is a release branch, does it have merge approval in the JIRA ticket.
* [ ] Extra checks if forced landing is requested
  * [ ] Review comments are sufficiently resolved, particularly by prior reviewers that requested changes.
  * [ ] No new NLT or valgrind warnings.  Check the classic view.
  * [ ] Quick-build or Quick-functional is not used.
* [ ] Fix the commit message upon landing. Check the standard [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments). Edit it to create a single commit. If necessary, ask submitter for a new summary.

### CODE REVIEW

File: src/common/btree_class.c [hunk 1]
```diff
@@ -1444,8 +1444,11 @@ iv_rec_string(struct btr_instance *tins, struct btr_record *rec, bool leaf,
 	void              *k;
 	uint64_t	key;
 
-	if (!leaf) {
-		key_rec = umem_off2ptr(&tins->ti_umm, rec->rec_node[0]);
+	if (!leaf && (tins->ti_root->tr_feats & BTR_FEAT_DIRECT_KEY)) {
```
Comment (Nasf-Fan) [inline src/common/btree_class.c:1447]: Why need to check "tins->ti_root->tr_feats & BTR_FEAT_DIRECT_KEY"? As my understand, for non-leaf node, nd[0] is always the header to trace current level records, in spite of whether it is direct key or not, right? Seems the API to_rec_string for other classes also have similar issue.

Comment (NiuYawei) [reply src/common/btree_class.c:1447]: For non-direct key, we use the 'rec_hkey' in current btr_record (see following iv_key_get() call). As for other tree classes, I think we don't support direct key for them yet.

**Review by Nasf-Fan — APPROVED** [review]
I see. Thanks for the explanation.

**Comment by NiuYawei** [comment]
Hit DAOS-12326 two times.

**Review by jolivier23 — DISMISSED** [review]

**Review by Nasf-Fan — DISMISSED** [review]

**Comment by NiuYawei** [comment]
Test failed for DAOS-11992

File: utils/run_test.sh [hunk 2]
```diff
@@ -200,6 +200,12 @@ if [ -d "/mnt/daos" ]; then
     COMP="UTEST_vea"
     run_test "${SL_PREFIX}/bin/vea_ut"
     run_test "${SL_PREFIX}/bin/vea_stress -d 60"
+    # regression test for DAOS-12256
```
Comment (jolivier23) [inline utils/run_test.sh:203]: probably should set COMP="UTEST_vea_debug" or something so the test gets categorized in Jenkins as a new test

**Review by jolivier23 — CHANGES_REQUESTED** [review]

**Review by Nasf-Fan — APPROVED** [review]

**Review by jolivier23 — APPROVED** [review]

**Pull request merged**